src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -558,6 +558,14 @@ void checkConstraintLanguageVersionAndRuntimeSupportForConversion(SyntaxNode syn
             }
 
             var syntax = (ExpressionSyntax)node.Syntax;
+            if (LocalRewriter.IsAllocatingRefStructCollectionExpression(node, targetType, elementType, Compilation))
+            {
+                diagnostics.Add(node.HasSpreadElements(out _, out _)
+                    ? ErrorCode.WRN_CollectionExpressionRefStructSpreadMayAllocate
+                    : ErrorCode.WRN_CollectionExpressionRefStructMayAllocate,
```

Comment:
Looks like if a user gets the former warning, then removes spreads they might still get the latter warning - isn't that confusing?

Reply:
It could def be inconvenient to address one warning and then have another warning pop up. However, I think giving a more precise answer here could require us to more deeply analyze the collection-expr, e.g. if the spread element were removed, would we be able to use the ROS optimization based on the type and other elements? Would we be able to use InlineArray?

The main goal of having multiple diagnostic IDS is to allow end users to choose a different policy for "allocating because of spreads" versus "allocating due to lack of runtime support". I think for the audience we expect to be consuming this warning, the current behavior is adequate.